src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToAsyncMethodsTests.cs

```diff
@@ -0,0 +1,2769 @@
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(int x, CancellationToken ct, string s) => Task.CompletedTask;
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_OverloadWithMultipleTokens()
+        {
+            return VerifyCS.VerifyAnalyzerAsync(@"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync();
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(CancellationToken c1, CancellationToken ct2) => Task.CompletedTask;
+}
+            ");
+        }
+
+        [Fact]
+        public Task CS_NoDiagnostic_OverloadWithMultipleTokensSeparated()
+        {
+            return VerifyCS.VerifyAnalyzerAsync(@"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync();
+    }
+    Task MethodAsync() => Task.CompletedTask;
+    Task MethodAsync(int x, CancellationToken c1, string s, CancellationToken ct2) => Task.CompletedTask;
+}
+            ");
+        }
+
+        #endregion
+
+        #region Diagnostic = C#
+
+        [Fact]
+        public Task CS_Diagnostic_Class_TokenDefault()
+        {
+            string originalCode = @"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await [|MethodAsync()|];
+    }
+    Task MethodAsync(CancellationToken c = default) => Task.CompletedTask;
+}
+            ";
+            string fixedCode = @"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync(ct);
+    }
+    Task MethodAsync(CancellationToken c = default) => Task.CompletedTask;
+}
+            ";
+            return VerifyCS.VerifyCodeFixAsync(originalCode, fixedCode);
+        }
+
+        [Fact]
+        public Task CS_Diagnostic_Class_TokenDefault_WithConfigureAwait()
+        {
+            string originalCode = @"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await [|MethodAsync()|].ConfigureAwait(false);
+    }
+    Task MethodAsync(CancellationToken c = default) => Task.CompletedTask;
+}
+            ";
+            string fixedCode = @"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    async void M(CancellationToken ct)
+    {
+        await MethodAsync(ct).ConfigureAwait(false);
+    }
+    Task MethodAsync(CancellationToken c = default) => Task.CompletedTask;
+}
+            ";
+            return VerifyCS.VerifyCodeFixAsync(originalCode, fixedCode);
+        }
+
+        [Fact]
+        public Task CS_Diagnostic_NoAwait()
+        {
+            string originalCode = @"
+using System.Threading;
+using System.Threading.Tasks;
+class C
+{
+    void M(CancellationToken ct)
+    {
+        [|MethodAsync()|];
```

Comment:
nit: as far as Location for the diagnostic, I would prefer just the method name rather than the entire invocation.
Two arguments for this:
1. It's the method chosen and not _all_ the arguments that are wrong.
1. For method invocations that span several lines, seeing a multi-line squiggle *feels* a bit overwhelming.

Reply:
> For method invocations that span several lines, seeing a multi-line squiggle feels a bit overwhelming.

Especially ones that take lambdas as arguments.

Reply:
@carlossanlop You likely would need to write language specific code to get to the syntax for member being invoked from `IInvocationOperation`. Ideally you would be able to get the underlying `IMemberReferenceOperation` for the member being invoked, but that is not the case today due to https://github.com/dotnet/roslyn/issues/26206.

Reply:
Sure I can do that. And thanks @mavasani for pointing out there's a bug that would prevent me from getting the underlying operation.